src/haven/instrument/robot.py

```diff
@@ -0,0 +1,124 @@
+import asyncio
+import logging
+
+from ophyd import Component as Cpt
+from ophyd import DynamicDeviceComponent as DCpt
+from ophyd import Device, EpicsMotor, EpicsSignal
+
+from .._iconfig import load_config
+from .device import aload_devices, make_device
+
+log = logging.getLogger(__name__)
+
+def transfer_sample(num_dios: int):
+    """Create a dictionary with robot sample device definitions.
+    For use with an ophyd DynamicDeviceComponent.
+    Parameters
+    ==========
+    num_dios
```

Comment:
These are not digital I/Os, so we should rename this variable to something more descriptive. ``num_samples`` would work.